src/test/ui/associated-const/defaults-cyclic-pass.rs

```diff
@@ -0,0 +1,35 @@
+// run-pass
```

Comment:
Ah nice! This test gives a good intuition about the second unresolved question. 🎉

Reply:
@jonas-schievink Since this is a rather ephemeral place, could you add a comment to the tracking issue re. the unresolved question around cycles with reference to associated consts and this test?

Reply:
Described in https://github.com/rust-lang/rust/issues/29661#issuecomment-502257533. Thanks!